lib/msf/core/exploit/remote/http/spip.rb

```diff
@@ -1,46 +1,111 @@
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'spip.php')
+      )
 
-  # Determine Spip version
-  #
-  # @return [Rex::Version] Version as Rex::Version
-  def spip_version
-    res = send_request_cgi(
-      'method' => 'GET',
-      'uri'    => normalize_uri(target_uri.path, "spip.php")
-    )
+      return unless res
 
-    return unless res
+      version = nil
 
-    version = nil
+      potential_sources = [
+        res.get_html_document.at('head/meta[@name="generator"]/@content')&.text,
+        res.headers['Composed-By']
+      ]
 
-    version_string = res.get_html_document.at('head/meta[@name="generator"]/@content')&.text
-    if version_string =~ /SPIP (.*)/
-      version = ::Regexp.last_match(1)
-    end
+      potential_sources.each do |text|
+        next unless text
+
+        if text =~ /SPIP\s(\d+(\.\d+)+)/
+          version = ::Regexp.last_match(1)
+          break
+        end
+      end
 
-    if version.nil? && res.headers['Composed-By'] =~ /SPIP (.*)/
-      version = ::Regexp.last_match(1)
+      return version ? Rex::Version.new(version) : nil
     end
 
-    if version.nil?
-      return nil
+    # Determine Spip plugin version by name
+    #
+    # @param [String] plugin_name Name of the plugin to search for
+    # @return [Rex::Version, nil] Version of the plugin as Rex::Version, or nil if not found
+    def spip_plugin_version(plugin_name)
+      res = send_request_cgi(
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'spip.php')
+      )
+
+      return unless res
+
+      # Check the Composed-By header for plugin version or config.txt URL
+      composed_by = res.headers['Composed-By']
+      return unless composed_by
+
+      # Case 1: Look for config.txt URL in the header
+      if composed_by =~ %r{(https?://[^\s]+/local/config\.txt)}i
+        config_url = ::Regexp.last_match(1)
+        vprint_status("Found config.txt URL: #{config_url}")
+
+        # Fetch and parse the config.txt file directly
+        config_res = send_request_cgi(
+          'method' => 'GET',
+          'uri' => config_url
+        )
+
+        if config_res&.code == 200
+          return parse_plugin_version(config_res.body, plugin_name)
+        end
+      end
+
+      # Case 2: Check for plugin version directly in Composed-By
+      composed_by.split(',').each do |entry|
+        if entry =~ /#{plugin_name}\((\d+(\.\d+)+)\)/
+          return Rex::Version.new(::Regexp.last_match(1))
+        end
+      end
+
+      # Case 3: Fallback to fetching /local/config.txt directly
+      vprint_status('No version found in Composed-By header. Attempting to fetch /local/config.txt directly.')
+      config_url = normalize_uri(target_uri.path, 'local', 'config.txt')
+      config_res = send_request_cgi(
+        'method' => 'GET',
+        'uri' => config_url
+      )
+
+      return parse_plugin_version(config_res.body, plugin_name) if config_res&.code == 200
```

Comment:
I think this should be done even when there is no `Composed-By` header

Reply:
Yes, that's what's done here, if we don't have the url in the `Composed-By` Header we check in `/local/config.txt` assuming the file exists.

Reply:
```ruby
      # Check the Composed-By header for plugin version or config.txt URL
      composed_by = res.headers['Composed-By']
      return unless composed_by
```

there is an early return if the header is absent :)

Reply:
Fixed https://github.com/rapid7/metasploit-framework/pull/19444/commits/289f47fac181048bca9124cac69d3e478d50ff7e